### PR TITLE
cmd/scollector: Identify the interface type via IF-MIB:ifType

### PR DESCRIPTION
**Pull request by kylebrandt**
Instead of string parsing - mostly.

### CODE REVIEW

File: cmd/scollector/collectors/snmp_ifaces.go [hunk 1]
```diff
@@ -41,15 +42,27 @@ func SNMPIfaces(cfg conf.SNMP) {
...
+	switch ifType {
+	case 6:
+		return metric
+	case 53:
+		return osNet + ".bond" + strings.TrimPrefix(metric, osNet)
+	case 135:
+		return osNet + ".virtual" + strings.TrimPrefix(metric, osNet)
+	case 131:
+		return osNet + ".tunnel" + strings.TrimPrefix(metric, osNet)
+	case 161:
```
Comment (captncraig) [inline cmd/scollector/collectors/snmp_ifaces.go]: You can combine this with the `53` case by using `fallthrough`.

File: cmd/scollector/collectors/snmp_ifaces.go [hunk 2]
```diff
@@ -64,11 +77,23 @@ func c_snmp_ifaces(community, host string) (opentsdb.MultiDataPoint, error) {
 	if err != nil {
 		return nil, err
 	}
+	t, err := snmp_subtree(host, community, ifType)
```
Comment (captncraig) [inline cmd/scollector/collectors/snmp_ifaces.go]: can we fix up some of the one letter variable names in here? k and v I don't mind but a,n,t should all be more descriptive. `snmpNames`, `snmpTypes` perhaps? `ifTypes`?

Comment (kylebrandt) [inline cmd/scollector/collectors/snmp_ifaces.go]: Sure, the tricky thing is that they are basically the "preprocessed"  versions of those, any suggestions?

Comment (captncraig) [inline cmd/scollector/collectors/snmp_ifaces.go]: perhaps `names` and `nameMap`

**Pull request merged**